core/trino-main/src/main/java/io/trino/server/ui/OAuthWebUiCookie.java

```diff
@@ -66,4 +67,13 @@ public static NewCookie delete(boolean secure)
                 secure,
                 true);
     }
+
+    private static String getName(boolean secure)
+    {
+        if (secure) {
+            // prefix according to: https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-05#section-4.1.3.1
+            return "__Secure-" + OAUTH2_COOKIE;
```

Comment:
`__Host-` is better option than `__Host-`
```
The __Secure- prefix makes a cookie accessible from HTTPS sites only. A HTTP site can not read or update a cookie if the name starts with __Secure-. This protects against the attack we earlier described, where an attacker uses a forged insecure site to overwrite a secure cookie.
The __Host- prefix does the same as the __Secure- prefix and more. A __Host--prefixed cookie is only accessible by the same domain it is set on. This means that a subdomain can no longer overwrite the cookie value.
```
https://www.sjoerdlangkemper.nl/2017/02/09/cookie-prefixes/
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#cookie_prefixes

Reply:
That's a good point, but `__Host-` cookie can be set only for the `/` path which is not the case here because we're setting it for the `/ui` only.